Use database storage and new web interfaces

pascalaubry opened the pull request.

pascalaubry committed:
Fixed ChessEvent edit modal

pascalaubry committed:
Fixed admin object creation workflow

pascalaubry committed:
Fixed tournament warning

pascalaubry committed:
Limit ChessEvent updates

pascalaubry committed:
Set #please-wait fixed

pascalaubry committed:
Fixed home icon size

pascalaubry committed:
Fixed admin event config page header

pascalaubry committed:
Fixed cards container header

pascalaubry committed:
Deleted unused file

pascalaubry committed:
Fixed cards container header

pascalaubry committed:
Fixed admin tournament list

pascalaubry committed:
Removed old admin header form

pascalaubry committed:
Deleted unneeded test

pascalaubry committed:
Finalized the admin rotator, timer and ChessEvent list pages
Minimized the openings of the Papi databases (added a lazy mode to class Loader)
Fixed misc layout details
Factorized admin columns buttons code
Increased the user screens ' update delay from 5 to 10'
Added the possibility to choose the number of columns of the cards layout
Forbid characrter ':' in screen uniq ids (used for family screens)

pascalaubry committed:
Removed unneeded write access to the Papi database (when reading illegal moves)

pascalaubry committed:
Hide the number of characters of ChessEvent passwords

pascalaubry committed:
Fixed a bug on screen creation

pascalaubry committed:
Use POST URLs for all user screen links
Use Bootstrap responsive styles wherever possible
When creating a screen, use the first tournament by alphabetical uniq_id order for the screen set
Removed the fade attribute to solve the right padding modal bug (https://stackoverflow.com/questions/32862394/bootstrap-modals-keep-adding-padding-right-to-body-after-closed)

pascalaubry committed:
Changed the title of players screens

pascalaubry committed:
Made the table headers sticky

pascalaubry committed:
Made the table headers sticky
Make the users choose their number of columns
Added tabs to the user screens page

pascalaubry committed:
Prevent screens from refreshing in rotators
Added classes *UserWebContext to factorize web code

pascalaubry committed:
Do not send static files on HTMX requests

pascalaubry committed:
Hide screen menus on rotators

pascalaubry committed:
Removed unwanted warning with private events on public screens

pascalaubry committed:
Moved user_selector to user_event_selector

pascalaubry committed:
Merged all user input classes to views_user_input.py
Moved rotator and login classes to UserScreenController

pascalaubry committed:
Moved _admin_render_index() back to AAdminController
Moved _event_login_needed to UserScreenController
Moved _user_render_event() to UserEventController

pascalaubry committed:
Renamed views_*_index to view_*

pascalaubry committed:
Simplified all the controllers by using web contexts
Fixed mouse pointer on screen menus
Fixed rotator icon
Fixed misc layout issues

pascalaubry committed:
Use web contexts in all controller methods
Fixed misc layout issues
Fixed bug on event deletion
Lazy load user events for the user events page

pascalaubry committed:
Removed unused old classes used with the INI format

pascalaubry committed:
Renamed all the new classes using the new database storage

pascalaubry committed:
Use cards for admin events page

pascalaubry committed:
Added extra control on rotator and screen visibility on user pages

pascalaubry committed:
Removed unused imports

pascalaubry committed:
Added image screens
Added /favicon.ico URL
Uniformized card colors
Fixed various layout bugs

pascalaubry committed:
Added a background color to image screens

pascalaubry committed:
Shortened event tabs titles

pascalaubry committed:
Fixed a bug on screen set edition

pascalaubry committed:
Removed unused constants and methods

pascalaubry committed:
Moved TMP_DIR back to common

pascalaubry committed:
Reorganized requirements.txt

pascalaubry committed:
Updated all the librairies
Replaced deprecated mysql-connector by mysql-connector-python
Upgraded Litestar to 2.11.0
Upgraded Jinja2 to 3.1.4

pascalaubry committed:
Renamed Screen.image to image_url
Renamed Screen.color to image_color

pascalaubry committed:
Fixed event loading in ChessEventEngine

pascalaubry committed:
Moved custom path to PapiWebConfig
Reorganized custom folder
Replaced Event.css by image_url and image_color

pascalaubry committed:
Reorganized custom folder

pascalaubry committed:
Removed unused methods

pascalaubry committed:
Protect custom folder read access

pascalaubry committed:
Factorized tooltip and modal code
Fixed background images and colors

pascalaubry committed:
Updated FRA logo
Removed useless macro params

pascalaubry committed:
Removed the link to GitHub in the copyright

pascalaubry committed:
Fixed screen rotation issue
Factorized template contexts

pascalaubry committed:
Removed useless JS

pascalaubry committed:
Improved screen and event cards by adding background images and colors

pascalaubry committed:
Fixed handicap time control color issue

pascalaubry committed:
Fixed handicap time control increment issue

Amaras commented on common/__init__.py:
```diff
@@ -0,0 +1,45 @@
+    return '#' + ''.join(f'{max(0, min(255, i)):02X}' for i in rgb)
+
+
+def format_timestamp_date_time(ts: float = None) -> str:
+    return datetime.strftime(datetime.fromtimestamp(ts if ts is not None else time.time()), '%Y-%m-%d %H:%M')
+
+
+def format_timestamp_date(ts: float = None) -> str:
+    return datetime.strftime(datetime.fromtimestamp(ts if ts is not None else time.time()), '%Y-%m-%d')
+
+
+def format_timestamp_time(ts: float) -> str:
+    return datetime.strftime(datetime.fromtimestamp(ts if ts is not None else time.time()), '%H:%M')
```
For each parameter: should be a `float | None` instead of a `float`

pascalaubry replied:
Fixed.

Amaras commented on common/__init__.py:
```diff
@@ -0,0 +1,45 @@
+def check_rgb_str(color: str) -> str:
+    rgb: RGB = hexa_to_rgb(color)
+    if rgb:
+        return rgb_to_hexa(hexa_to_rgb(color))
```
Why recompute?

pascalaubry replied:
Fixed.

Amaras commented on chessevent/event_selector.py:
```diff
@@ -1,28 +1,27 @@
-        self.__silent = True  # verbose on the first call only
+    @staticmethod
+    def run() -> bool:
+        events: List[Event] = EventLoader.get(request=None, lazy_load=True).events_sorted_by_name
```
We should get rid of the `typing.List` annotation, and use the builtin `list`

pascalaubry replied:
Fixed (only occurrence).

Amaras commented on data/loader.py:
```diff
@@ -0,0 +1,194 @@
+        self._current_public_events: list[Event] | None = None
+        self._coming_public_events: list[Event] | None = None
+
+    @staticmethod
+    def get(request: HTMXRequest | None, lazy_load: bool):
+        if not request:
+            return EventLoader(lazy_load=lazy_load)
+        event_loader: EventLoader = request.state.get('event_loader')
+        if event_loader and lazy_load:
+            return event_loader
+        event_loader = EventLoader(lazy_load=False)
+        request.state['event_loader'] = event_loader
+        return event_loader
+
```
This feels more like a `classmethod`, rather than a `staticmethod`.

pascalaubry replied:
Fixed.

Amaras commented on data/loader.py:
```diff
@@ -0,0 +1,194 @@
+            self._loaded_stored_events_by_id[uniq_id] = self._load_stored_event(uniq_id)
+            return self._loaded_stored_events_by_id[uniq_id]
+
+    @property
+    def event_uniq_ids(self) -> list[str]:
+        if self._event_uniq_ids is None:
+            self._event_uniq_ids = [
+                file.stem for file in PapiWebConfig().event_path.glob(f'*.{PapiWebConfig().event_ext}')
+            ]
+        return self._event_uniq_ids
```
Maybe look into `functools.cached_property`

pascalaubry replied:
This obviously needs to be used in most of the new classes.
Keep this thread open.

pascalaubry replied:
Fixed (used wherever possible).

Amaras commented on export.py:
```diff
@@ -65,29 +63,34 @@ def build_exe():
+    yml_dir: Path = Path('.') / 'database' / 'yml'
+    files += [
+        file for file in yml_dir.glob('*.yml')
+    ]
```
```suggestion
    files += list(yml_dir.glob('*.yml'))
```

Amaras commented on web/server_engine.py:
```diff
@@ -42,24 +43,35 @@ def __init__(self):
         )
-        uvicorn.run(app, host=self._config.web_host, port=self._config.web_port, log_level='info',)
+        # This code is intended to check the uniformity of the paths and names used for the application URLs
+        url_map: dict[str, list[str]] = {}
+        for route in app.routes:
+            for handler in route.route_handlers:
+                if handler.name:
+                    if handler.name not in url_map:
+                        url_map[handler.name] = []
+                    url_map[handler.name].append(route.path)
+        for name in sorted(url_map.keys()):
+            logger.debug(f'{name}: {url_map[name]}')
+        uvicorn.run(app, host=papi_web_config.web_host, port=papi_web_config.web_port, log_level='info', )
```
A `defautdict` would remove the line 69 check.

pascalaubry replied:
Fixed.

Amaras commented on web/settings.py:
```diff
@@ -9,17 +9,29 @@
-from web.views_admin import AdminController
+from common.papi_web_config import PapiWebConfig
+from web.views import IndexController
+from web.views_admin import AdminIndexController
 from web.views_admin_chessevent import AdminChessEventController
 from web.views_admin_event import AdminEventController
-from web.views_user import UserController
+from web.views_admin_family import AdminFamilyController
+from web.views_admin_rotator import AdminRotatorController
+from web.views_admin_screen import AdminScreenController
+from web.views_admin_timer import AdminTimerController
+from web.views_admin_tournament import AdminTournamentController
+from web.views_background import BackgroundController
+from web.views_user import UserIndexController
+from web.views_user_event import UserEventController
+from web.views_user_screen import UserScreenController
+from web.views_user_screen_set import UserScreenSetController
+from web.views_user_tournament import UserCheckInController, UserIllegalMoveController, UserResultController, \
+    UserDownloadController
```
All those are `Controller`s.
Maybe you should put them in a `web.controllers` module?
Also, you should probably have `web.controllers.user` and `web.controllers.admin` submodules, because all those `views_admin_*` and `views_user_*` seem like a bad organization.

pascalaubry replied:
Feel the same, keep this thread open.

pascalaubry replied:
Added sub-modules web.controllers, web.controllers.admin and web.controllers.user and renamed the classes.

pascalaubry committed:
Fixed issue on the modification of allow_results_deletion_on_input_screens

pascalaubry committed:
Added a JS confirmation when leaving the page (to prevent from accidental F5)

pascalaubry committed:
Replaced Escape handling by an exit button on screens

pascalaubry committed:
Fixed user background screen display

pascalaubry committed:
Left pad family screens' uniq_id to 3 digits to respect alphabetical sorting

pascalaubry committed:
Fixed tooltips on the index page

pascalaubry committed:
Fixed background images on user pages

pascalaubry committed:
Fixed bug in ScreenAdminWebContext

pascalaubry committed:
Handle Escape keys on all admin modals

pascalaubry committed:
Added background images to family and rotator admin cards

pascalaubry committed:
Fixed bug in ScreenType.icon_str

pascalaubry committed:
Improved admin rotator card

pascalaubry committed:
Fixed show_menus rotator option modification

pascalaubry committed:
Improved event config page

pascalaubry committed:
Removed body right padding when loading body (needed after closing modals)

pascalaubry committed:
Fixed too long line

pascalaubry committed:
Restored the possibility to use a default menu label for screens and families

pascalaubry committed:
Go back to the correct list of passed/current/comming events when leaving an event page

pascalaubry committed:
Go back to the correct list of passed/current/comming events when leaving an event page

pascalaubry committed:
Enlightened the error background image

pascalaubry committed:
Return to the original tab when cancelling an event creation

pascalaubry committed:
Added an admin page for event messages

pascalaubry committed:
Fixed screen issues in bre-jeunes-2024.yml

pascalaubry committed:
Send the reload button to the right tab on the admin event lists

pascalaubry committed:
Prevent guys from deleting an event with the same name twice in a minute ;-)

pascalaubry committed:
Definitively prevent guys from deleting an event with the same name twice in a minute ;-)

pascalaubry committed:
Fixed checkboxes input on color edition

pascalaubry committed:
Improved event messages display

pascalaubry committed:
Removed unused variable

pascalaubry committed:
Added a file picker to ease the selection of background images

Amaras commented on data/loader.py:
```diff
@@ -0,0 +1,194 @@
+        request.state['event_loader'] = event_loader
+        return event_loader
+
+    def clear_cache(self, event_uniq_id: str = None):
```
```suggestion
    def clear_cache(self, event_uniq_id: str | None = None):
```

pascalaubry replied:
Fixed.

pascalaubry committed:
Present files before dirs on the background image picker
Added a tester for background images
Added a few background images

pascalaubry committed:
Present files before dirs on the background image picker
Added a tester for background images
Added a few background images

pascalaubry committed:
Fixed a bug on rotator cloning

pascalaubry committed:
Fixed a bug on timer creation

pascalaubry committed:
Added sortable and jstree to the export

pascalaubry committed:
Added sortable and jstree to the export
Release 2.4rc4

pascalaubry committed:
Removed unused JS code

pascalaubry committed:
Added one more background image for Avoine
Set nice (?) colors to pre-loaded events

pascalaubry committed:
Added jsTree and SortableJS version to the admin config page

pascalaubry committed:
Fixed a bug on events, screens, families and rotators visibility

pascalaubry committed:
Fixed event cards (number of messages)

Amaras commented on common/papi_web_config.py:
```diff
@@ -42,7 +31,7 @@
 @singleton
 class PapiWebConfig:
```
Couldn't all the properties not using the reader be class attributes instead?
I don't feel like properties are the right way to express constants.
Properties are meant to be used to do some work in addition to just returning instance attributes.

pascalaubry replied:
Indeed, fixed.

pascalaubry replied:
Mark unresolved until @singleton is removed on all classes.

pascalaubry replied:
@singleton replaced by metaclass=Singleton everywhere.

Amaras commented on data/loader.py:
```diff
@@ -0,0 +1,194 @@
+            ]
+        return self._event_uniq_ids
+
+    @property
+    def stored_events_by_id(self) -> dict[str, StoredEvent]:
+        if self._stored_events_by_id is None:
+            self._stored_events_by_id: dict[str, StoredEvent] = {}
+            for uniq_id in self.event_uniq_ids:
+                self._stored_events_by_id[uniq_id] = self.load_stored_event(uniq_id)
+        return self._stored_events_by_id
```
Feels like a `functools.cached_property`.
The pattern to recognised a cached property is the following:
```py
@property
def something(self) -> T: # T is a type
	if self._something is None:
		# initialize self._something
	return self._something
```

pascalaubry replied:
Yes, I need to make a complete review of the new classes for this (closing this conversation since one is still opened on the same subject).

Amaras commented on data/loader.py:
```diff
@@ -0,0 +1,194 @@
+    @property
+    def public_events(self) -> list[Event]:
+        if self._public_events is None:
+            self._public_events = sorted([
+                event for event in self.events_by_id.values()
+                if event.public
+            ], key=lambda event: event.name)
```
Should be a generator expression. I think `sorted()` accepts any iterable:
```suggestion
            self._public_events = sorted((
                event for event in self.events_by_id.values()
                if event.public
            ), key=lambda event: event.name)
```

Or even (formatting could be better):
```suggestion
            self._public_events = sorted(filter(lambda event: event.public, self.events_by_id.values()), key = lambda  event: event.name)
```

Or better yet:
```suggestion
	self._public_events = sorted(filter(attrgetter('public'), self.events_by_id.values(), key=attrgetter('name'))
```

pascalaubry replied:
Fixed (your last suggestion was missing a parenthesis).
Will be even nicer later with cached_property.

Amaras commented on database/store.py:
```diff
@@ -1,31 +1,27 @@
+    date_str: str | None = field(default=None)
+    time_str: str | None = field(default=None)
+    text_before: str | None = field(default=None)
+    text_after: str | None = field(default=None)
```
```suggestion
    order: int | None = None
    date_str: str | None = None
    time_str: str | None = None
    text_before: str | None = None
    text_after: str | None = None
```

Amaras commented on database/store.py:
```diff
@@ -35,114 +31,143 @@ class StoredChessEvent:
-    part: int | None
-    parts: int | None
-    number: int | None
+    last_update: float = field(default=0.0)
```
```suggestion
    last_update: float = 0.0
```

Amaras commented on database/store.py:
```diff
@@ -35,114 +31,143 @@ class StoredChessEvent:
+    results_tournament_ids: list[int] = field(default_factory=list[int])
     stored_screen_sets: list[StoredScreenSet] = field(default_factory=list[StoredScreenSet])
+    last_update: float = field(default=0.0)
+    public: bool = field(default=True)
```
```suggestion
    last_update: float = 0.0
    public: bool = True
```

Amaras commented on database/store.py:
```diff
@@ -35,114 +31,143 @@ class StoredChessEvent:
     parts: int | None
     number: int | None
+    public: bool = field(default=True)
+    last_update: float = field(default=0.0)
```
```suggestion
    public: bool = True
    last_update: float = 0.0
```

Amaras commented on database/store.py:
```diff
@@ -35,114 +31,143 @@ class StoredChessEvent:
+    screen_ids: list[int] | None
     delay: int | None
+    show_menus: bool | None
+    public: bool = field(default=True)
```
```suggestion
    public: bool = True
```

Amaras commented on database/store.py:
```diff
@@ -35,114 +31,143 @@ class StoredChessEvent:
+    stop: float
+    public: bool = field(default=False)
     path: str | None = field(default=None)
-    css: str | None = field(default=None)
+    background_image: str | None = field(default=None)
+    background_color: str | None = field(default=None)
     update_password: str | None = field(default=None)
     record_illegal_moves: int | None = field(default=None)
-    allow_deletion: bool | None = field(default=None)
+    allow_results_deletion_on_input_screens: bool | None = field(default=None)
+    version: str | None = field(default=None)
+    timer_colors: dict[int, str | None] = field(default=None)
+    timer_delays: dict[int, int | None] = field(default=None)
+    last_update: float = field(default=0.0)
```
Why not simply use `attribute: T = default` instead of `attribute: T = field(default=default)`? (when applicable, of course)

pascalaubry replied:
Yeah, why not? Fixed ;-)

Amaras commented on export.py:
```diff
@@ -4,17 +4,15 @@
 DATA_DIR: Path = Path('export-data')
-CUSTOM_DIR: Path = Path('custom')
-basename: str = f'papi-web-{PAPI_WEB_VERSION}'
+basename: str = f'papi-web-{PapiWebConfig().version}'
```
If you use class attributes, you don't need to initialize the singleton until absolutely necessary

pascalaubry replied:
Fixed for all the usages of PapiWebConfig.

Amaras commented on data/event.py:
```diff
@@ -1,305 +1,535 @@
 
 logger: Logger = get_logger()
 
+event_last_load_date_by_uniq_id: dict[str, float] = {}
```
Are you sure you want to use a global variable for this?

pascalaubry replied:
Well, couldn't find a better solution to replace the tmp files used previously.
Any other idea?

Amaras commented on data/event.py:
```diff
@@ -1,305 +1,535 @@
+
+    @property
+    def players_number(self) -> int:
+        return sum([len(tournament.players_by_name_with_unpaired) for tournament in self.tournaments_by_id.values()])
```
Use generator expressions
```suggestion
        return sum((len(tournament.players_by_name_with_unpaired) for tournament in self.tournaments_by_id.values()))
```

Amaras commented on data/event.py:
```diff
@@ -1,305 +1,535 @@
+    def public(self) -> bool:
+        return self.stored_event.public
+
+    @property
+    def tournaments_sorted_by_uniq_id(self) -> list[Tournament]:
+        if self._tournaments_sorted_by_uniq_id is None:
+            self._tournaments_sorted_by_uniq_id = sorted(
+                self.tournaments_by_id.values(), key=lambda tournament: tournament.uniq_id)
+        return self._tournaments_sorted_by_uniq_id
```
Feels like a `functools.cached_property`

pascalaubry replied:
Will also fix it soon.

pascalaubry replied:
Fixed.

Amaras reviewed (changes requested):
I'm missing at least 50 files, but here's a first review.

204 files is too much to review quickly, I recommend publishing a pre-release version 2.4-web for beta testing, and once we can review everything properly for the merge, we can probably release version 2.4.0.

pascalaubry committed:
Updated most of the documentation files (work in progress)

pascalaubry committed:
Updated the documentation of the database

Amaras commented on docs/81-urls.md:
```diff
@@ -2,6 +2,9 @@
 
 # Papi-web - Annexe technique : URLs utilisées par le serveur web
 
+> [!CAUTION]
+> Cette page décrivant les URLs utilisées par l'application est obsolète.
+
```
I agree that the URLs will be obsolete anyway because of the reorganization, but I still think public URLs are better.
In particular, screen readers don't necessarily detect changes if the URL doesn't change.

pascalaubry replied:
Not sure to understand "screen readers don't necessarily detect changes if the URL doesn't change".

Amaras replied:
Basically this comment on Reddit: don't reinvent the wheel: https://www.reddit.com/r/htmx/comments/ti8vnj/comment/i2ukfct.
While there is a possibility to notify a screen reader of a change, the easier method of notifying a screen reader user of a change is changing the page (and the URL)

Amaras commented on docs/02-v2-news.md:
````diff
@@ -13,35 +13,26 @@ Papi-web est désormais livré sous la forme d'un exécutable Windows autonome.
-[tournament]
-filename = amical-20230617
-```
+- À partir de la version 2.4, la configuration des évènements est réalisée à l'aide d'une interface web qui ne nécessite aucune compétence informatique.
````
```suggestion
À partir de la version 2.4, la configuration des évènements est réalisée à l'aide d'une interface web qui ne nécessite aucune compétence informatique avancée.
```

Amaras reviewed (commented):
Thank you for the documentation update.

We definitely need to discuss the disappearance of the public URLs, which I definitely don't agree with.

pascalaubry committed:
Removed useless double computation
Set correct argument types

pascalaubry committed:
Removed Any.List usage

pascalaubry committed:
Moved EventLoader from static to class method

pascalaubry committed:
Simplify list usage

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Use defaultdict to simplify

pascalaubry committed:
Fixed argument type

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Replaced properties by class attributes

pascalaubry committed:
Added missing method

pascalaubry committed:
Used an iterator instead of a list in EventLoader.public_events()

pascalaubry committed:
Initialize fields without method field()

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Initialize fields without method field()

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Initialize fields without method field()

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Initialize fields without method field()

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Initialize fields without method field()

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Initialize fields without the field method

pascalaubry committed:
Removed all the useless PapiWebConfig instanciation

pascalaubry committed:
Used generators instead of lists

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Have a little consideration for newbye users ;-)

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>

pascalaubry committed:
Merge remote-tracking branch 'origin/db' into db

Amaras commented on common/__init__.py:
```diff
@@ -0,0 +1,45 @@
+    return datetime.strftime(datetime.fromtimestamp(ts if ts is not None else time.time()), '%Y-%m-%d')
+
+
+def format_timestamp_time(ts: float) -> str:
```
Still missing the `ts: float | None` annotation

pascalaubry replied:
IMO None is never passed, closing.

pascalaubry replied:
Fixed.

pascalaubry committed:
Use metaclass=Singleton for PapiWebConfig to allow the usage of the class attributes

pascalaubry committed:
Fixed typo

pascalaubry committed:
Use @cached_property wherever possible

pascalaubry committed:
Used an iterator instead of a list in Event.public_rotators_sorted_by_uniq_id()

pascalaubry committed:
Removed function singleton()

pascalaubry committed:
Fixed argument type

pascalaubry committed:
Fixed family menu_text for automatically created events

pascalaubry committed:
Fixed nested tooltips

pascalaubry committed:
Added sub-modules web.controllers, web.controllers.admin and web.controllers.user and renamed the classes

pascalaubry committed:
Moved web/controllers.BackgroundWebContext.inline_image_url to common.background.BackgroundUtils

pascalaubry committed:
Updated all librairies

pascalaubry committed:
Fixed screen menu labels for auto-generated event domloup-fide

pascalaubry committed:
Merge branch 'dev' into db

pascalaubry committed:
Added comments to main web context and controller classes

pascalaubry committed:
Finalized 2.4rc4

pascalaubry committed:
Removed empty file

pascalaubry committed:
Added comments to SQLite database classes

Amaras committed:
Added documentation to the code in chessevent/ and common/

pascalaubry committed:
Fixed typos

pascalaubry committed:
Moved CHESSEVENT_URL from ChessEventSession to PapiWebConfig

pascalaubry committed:
Removed namespace BackgroundUtils

pascalaubry committed:
Removed PapiWebConfig useless instanciation

pascalaubry committed:
Removed .ini old mention

pascalaubry committed:
Fixed iterator usage

pascalaubry committed:
Finalized 2.4rc5

pascalaubry committed:
Finalized 2.4rc5

pascalaubry committed:
Finalized 2.4rc5

pascalaubry committed:
Removed old ini files in /export-data/events

pascalaubry committed:
Fixed Iterator import
Finalized 2.4rc6

pascalaubry committed:
Made the ChessEvent engine print a clear message when the Papi folder does not exist

pascalaubry committed:
Added the context when logging an event message

pascalaubry committed:
Fixed default background color checkbox label

pascalaubry committed:
Removed forgotten debug output

pascalaubry committed:
Fixed the display of the current screen in screen menus

pascalaubry committed:
Fixed a bug on screen set name

pascalaubry committed:
Fixed a bug in screen updates

pascalaubry committed:
restored first and last parameters for players screens families

pascalaubry committed:
Fixed a bug on the selection of families for rotators

pascalaubry committed:
Made the selection of families for rotators easier

pascalaubry committed:
Remove the copyright on image screens

pascalaubry committed:
Fixed event cloning

pascalaubry committed:
Also clone event properties path and update_password

pascalaubry committed:
Fixed bug on illegal moves actions (some remain)

pascalaubry committed:
Finalized 2.4rc7

pascalaubry committed:
Drastically increased the performance on user/admin pages by really using lazy loading

pascalaubry committed:
Finalized 2.4rc8

Amaras committed:
Code documentation pass

pascalaubry committed:
Made parameter data optional for WebContext.__init__()
Allow passing exceptions to AbstractController.redirect_error()
Added method AbstractController.get_if_modified_since()
Added field now_http_date to basic default template context

pascalaubry committed:
Fixed bug on ChessEvent connection creation

pascalaubry committed:
Use method GET for URL /background

pascalaubry committed:
Finalized 2.4rc9

pascalaubry committed:
Merge remote-tracking branch 'origin/db' into db

# Conflicts:
#	web/controllers/index_controller.py

Amaras committed:
Reworked erroneous type annotations

pascalaubry committed:
Fixed forward reference issue

pascalaubry committed:
Merge remote-tracking branch 'origin/db' into db

# Conflicts:
#	data/screen.py

pascalaubry committed:
Unfixed forward reference issue

pascalaubry committed:
Fixed forward reference issue

pascalaubry committed:
Fixed too lazy event loading

pascalaubry committed:
Get event loaders only when needed on admin pages

pascalaubry committed:
Removed obsolete reference to ini files

pascalaubry committed:
Finalized 2.4rc10

pascalaubry committed:
Split user.html to user_index.html and user_event.html
Use appropriate methods for all /user/* URLs
Use header If-Modified-Since to refresh pages

pascalaubry committed:
Split admin.html to admin_index.html and admin_event.html
Use appropriate methods for all /admin/event/* URLs
Use appropriate methods for all /admin/tournament/* URLs

pascalaubry committed:
Removed unused method

pascalaubry committed:
Removed empty line

pascalaubry committed:
Unified template names

pascalaubry committed:
Fixed details on tournament admin pages

pascalaubry committed:
Use appropriate methods for all /admin/family/* URLs

pascalaubry committed:
Removed unused import

pascalaubry committed:
Removed useless parameter

pascalaubry committed:
Use appropriate methods for all /admin/rotator/* URLs

pascalaubry committed:
Fixed URL on rotators page

pascalaubry committed:
Sort tournaments alphabetically on admin results screen cards

pascalaubry committed:
Moved Admin[Family|Screen]Controller._get_tournament_options() to AdminEventWebContext
Use appropriate methods for all /admin/screen/* URLs
Fixed tournament selection of results screens

pascalaubry committed:
Fixed rotators management

pascalaubry committed:
Fixed rotators management

pascalaubry committed:
Fixed admin controllers error handling

pascalaubry committed:
Use appropriate methods for all /admin/timer/* URLs

pascalaubry committed:
Use appropriate methods for all /admin/chessevent/* URLs

pascalaubry committed:
Fixed admin event messages page

pascalaubry committed:
Fixed wrong typing

pascalaubry committed:
Create list as a literal

pascalaubry committed:
Fixed htmx_user_input_screen_* method names

pascalaubry committed:
Fixed bug on event creation

pascalaubry committed:
Added F35046 image

pascalaubry committed:
Allowed empty dicts in YML files

pascalaubry committed:
Hide passwords on ChessEvent cards

pascalaubry committed:
Hide passwords on ChessEvent cards

pascalaubry committed:
Hide passwords on ChessEvent cards

pascalaubry committed:
Force float typing for players' points

pascalaubry committed:
Force event number of players on boards screen before pairing the first round

pascalaubry committed:
Improved F35046 image

pascalaubry committed:
Added liffre default event
Fixed typo in avoine YML file

pascalaubry committed:
Fixed liffre-2024.yml
Fixed family names
Improved family cards
Fixed typo in family edit modal

pascalaubry committed:
Added title to admin_config.html

pascalaubry committed:
Added an admin tab for archived events
Improved error handling
Finalized 2.4rc11

pascalaubry committed:
Fixed the background image for the archives admin tab

pascalaubry committed:
Made ChessEvent/FFE engines handle event database deletion/renaming

pascalaubry committed:
Fixed modal_script() comment

pascalaubry committed:
Fixed double user_timer.js import

pascalaubry committed:
Improved logging info for illegal moves

pascalaubry committed:
Fixed all refresh issues on user screens

pascalaubry committed:
Added forfeit results on admin result modal

pascalaubry committed:
Added Papi template 3.3.8

pascalaubry committed:
Improved error handling

pascalaubry committed:
Redirect to / on error (eg database deletion or renaming) with no confirmation

pascalaubry committed:
Updated libraries

pascalaubry committed:
Finalized 2.4rc12

pascalaubry committed:
Redirect to / on error (e.g. database deletion or renaming) with no confirmation
Added the HTMX preload extension
Preload all GET requests when possible

pascalaubry committed:
Added precisions ont the difference between maintainers and contributors
Removed obsolete sandbox files

pascalaubry committed:
Added LICENSE.md

pascalaubry committed:
Set the AGPL

pascalaubry committed:
Set the AGPL

pascalaubry committed:
Deleted LICENSE

pascalaubry committed:
Added the copyright

pascalaubry committed:
Added comments to blocks
Reorganized user modals

pascalaubry committed:
Fixed event creation button

pascalaubry committed:
Fixed admin pages footer

pascalaubry committed:
Fixed admin-chessevent-modal URL path
Set caching wherever possible to improve preloading
Fixed #please-wait visibility
Cleaned admin controllers and HTML pages

pascalaubry committed:
Finalized 2.4rc13

pascalaubry committed:
Added a few guidelines to 83-dev-setup.md

pascalaubry committed:
Fixed image link in 83-dev-setup.md

pascalaubry committed:
Typo

pascalaubry committed:
Use smooth transitions on all page by setting hx-swap="transition:true" to the <body> tag

pascalaubry committed:
Removed useless (at this time) multi-swap HTMX extension

pascalaubry committed:
Updated libraries (except httpx)

pascalaubry committed:
removed user_columns_buttons.html

Amaras commented on COPYRIGHT.md:
```diff
@@ -0,0 +1,35 @@
+- faire référence au produit d'origine (Papi-web) ;
+- ne pas laisser d’ambiguïté sur l'origine du produit, qui pourrait laisser penser que vous en êtes l'auteur·ice ;
+
+Le terme distribution inclus ici toute livraison, même à un client unique, ainsi que la fourniture de services (hébergement, mode SaaS).
```
```suggestion
Le terme distribution inclut ici toute livraison, même à un client unique, ainsi que la fourniture de services (hébergement, mode SaaS).
```

Amaras reviewed (dismissed):
I can't review everything in this PR, so I'm not gonna delay it more.
Please make sure the next PRs are way smaller (270 files changed is definitely too much when it's not a pure documentation PR).

pascalaubry committed:
Added missing Papi files

pascalaubry committed:
Moved 02-v2-news.md to 02-roadmap.md

pascalaubry committed:
Update COPYRIGHT.md

Co-authored-by: Sammy Plat <sammy@sammyplat.fr>
